src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -69,20 +69,7 @@ public IThenConfiguration<IVoidConfiguration> NumberOfTimes(int numberOfTimes)
         public virtual IAfterCallConfiguredConfiguration<IVoidConfiguration> Throws(Func<IFakeObjectCall, Exception> exceptionFactory)
         {
             this.AddRuleIfNeeded();
-            this.RuleBeingBuilt.UseApplicator(call =>
-            {
-                Exception exceptionToThrow;
-                try
-                {
-                    exceptionToThrow = exceptionFactory(call);
-                }
-                catch (Exception ex) when (!(ex is FakeConfigurationException))
-                {
-                    throw new UserCallbackException(ExceptionMessages.UserCallbackThrewAnException("Exception factory"), ex);
-                }
-
-                throw exceptionToThrow;
-            });
+            this.RuleBeingBuilt.UseApplicator(call => throw exceptionFactory(call));

```

Comment:
I never noticed before, but throwing the return value does look weird. It's correct, but weird-looking!

Reply:
What's weird about it?

Reply:
Just your comment earlier about how we throw the thing that's returned, but if the method throws instead, we let it go and it all works out.
Maybe it's not weird. Maybe _I'm_ weird!